### PR TITLE
Fix/#42 fix installation files

### PR DESCRIPTION
**Pull request by GewoonMaarten**
In this pull request we changed the following to the repository and build process:

- Changed every component.tsx to index.tsx, this allows us to import directly (e.g. `
import FormControl from '@gemeente-denhaag/formcontrol'` instead of `@gemeente-denhaag/formcontrol/FormControl'`).

- Every component is build with the typescript transpiler to the target es5
- Each storybook story is moved to their component folder in `src/components/` - This was necessary since each component now outputs javascript and storybook cannot infer types from this to autogenerate documentation.
- Added react as peer dependency to every component
- Storybook stories refer to local packages, so make sure to do a `yarn build` before `yarn storybook`
- Upgraded react and storybook to their latest version (in all package.jsons)

Also closes PR #114, #113 and #131

**Testing**
We have tested this using a verdaccio local registry on the PWA master and development branches.
This can be repeated by installing and running verdaccio on the default port, navigate to http://localhost:4873 and follow instructions to login on the registry, username, password and email do not matter.
If you have used verdaccio before, remove the content in the verdaccio/storage folder.

A lerna publish will create and push tags to your remote.
In order to test this without publishing tags and commits to the denhaag origin, create a dummy github repository and set your git remote origin to it temporarily.
Make sure the tags that are created on publish don't exist yet in the local git registry, otherwise remove them locally: `git tag | xargs git tag -d`

Run the following command to publish a new version to the local registry (on a separate branch e.g. fix/test):
`lerna publish --registry http://localhost:4873 --allow-branch fix/test --skip-git`

Now you can test the packages in a new React project, or test it in the PWA. Either way make sure you specify the registry with `--registry http://localhost:4873`.
In the PWA yarn upgrade the dependency of @gemeente-denhaag/denhaag-component-library (with the registry flag), and correct the imports.

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
 module.exports = {
   stories: [
-    '../src/stories/*.stories.mdx',
-    '../src/stories/*.stories.@(js|jsx|ts|tsx)',
-    '../src/stories/*/*.stories.@(js|jsx|ts|tsx)',
+    // '../src/stories/*.stories.mdx',
+    // '../src/stories/*.stories.@(js|jsx|ts|tsx)',
+    // '../src/stories/*/*.stories.@(js|jsx|ts|tsx)',
+    './stories/*.stories.mdx',
```
Comment (ogjtech) [inline .storybook/main.js:6]: Won't this break? The stories folder is also in the `src` folder

Comment (ogjtech) [reply .storybook/main.js:6]: Just noticed you moved all the files into the .storybook folder. Is this recommended somewhere? I'm not sure whether we want to introduce this change into the project structure like this.

**Comment by GewoonMaarten** [comment]
Squash and merge this PR to keep the history clean

File: .storybook/main.js (hunk 1 above)
Comment (DanielvanVliet) [reply .storybook/main.js:3]: The reason for this is now explained in the PR description, namely that every story requires access to typescript files to generate documentation.  In our build we no longer expose ts files, furthermore the storybook file watchers were not working anymore, and storybook was not reloaded on updates in stories.tsx or component source code.

File: package.json [hunk 2]
```diff
@@ -1,30 +1,25 @@
...
-  "engines": {
-    "node": ">=10",
-    "yarn": "^1.22.0"
-  },
   "scripts": {
-    "build": "microbundle-crl --no-compress --format modern,cjs",
+    "bootstrap": "lerna bootstrap",
+    "build": "lerna run build",
     "build-storybook": "build-storybook",
+    "clean": "lerna run clean --parallel",
```
Comment (bddjong) [inline package.json:22]: Didn't the use of --parallel crash your machine @GewoonMaarten ?

**Comment by bddjong** [comment]
There's a lot of discoveries in how we can test these packages locally, and a lot of changes in how it builds. Is there anything we can document in the README.md or CONTRIBUTING.md?

File: package.json (hunk 2 above)
Comment (GewoonMaarten) [reply package.json:22]: `clean` ran just fine, but `build` with `--parallel` crashed my laptop.

File: src/components/TextField/src/TextField.stories.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import { Meta, Story } from "@storybook/react";
-import pkg from "../../components/TextField/package.json";
-import TextField, { TextFieldProps } from "../../components/TextField";
+import pkg from "../package.json";
+import TextField, { TextFieldProps } from "../../TextField";
```
Comment (DanielvanVliet) [inline src/components/TextField/src/TextField.stories.tsx:4]:
```suggestion
import TextField, { TextFieldProps } from ".";
```

Comment (GewoonMaarten) [reply src/components/TextField/src/TextField.stories.tsx]: fixed

**Comment by bddjong** [comment]
Plz never make me do this again

**Review by bddjong — APPROVED** [review]

File: .storybook/main.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
 module.exports = {
   stories: [
-    '../src/stories/*.stories.mdx',
-    '../src/stories/*.stories.@(js|jsx|ts|tsx)',
-    '../src/stories/*/*.stories.@(js|jsx|ts|tsx)',
+    './stories/*.stories.mdx',
+    '../src/components/*/*.stories.mdx',
+    '../src/components/*/src/*.stories.@(js|jsx|ts|tsx)'
```
Comment (joostvanviegen) [inline .storybook/main.js:5]: Do we actually have files residing at a location identified by this regex?

File: lerna.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
   "version": "independent",
   "command": {
       "publish": {
-        "allowBranch": "development"
+        "allowBranch": "master"
```
Comment (joostvanviegen) [inline lerna.json:7]: The reasoning behind allowing development was the inability to push to master, has this been changed?

**Review by ogjtech — APPROVED** [review]

File: src/components/List/src/index.tsx [hunk 6]
```diff
@@ -34,10 +34,17 @@ export const List: React.FC<ListProps> = (props: ListProps) => {
...
 export * from "@gemeente-denhaag/listitemtext";
 export * from "@gemeente-denhaag/listsubheader";
+
+export default List;
+// export { ListItem, ListItemProps };
+// export { ListItemAvatar, ListItemAvatarProps };
+// export { ListItemIcon, ListItemIconProps };
+// export { ListItemSecondaryAction, ListItemSecondaryActionProps };
+// export { ListItemText, ListItemTextProps };
+// export { ListSubheader, ListSubheaderProps };
```
Comment (joostvanviegen) [inline src/components/List/src/index.tsx:50]: I think we should remove these commented lines

File: src/components/Pickers/src/index.tsx [hunk 7]
```diff
@@ -0,0 +1,11 @@
+// import DatePicker, { DatePickerProps } from "./DatePicker/DatePicker";
```
Comment (joostvanviegen) [inline src/components/Pickers/src/index.tsx:1]: These commented lines should be removed if we are not using them

File: src/components/StepLabel/src/index.tsx [hunk 8]
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import MaterialStepLabel from "@material-ui/core/StepLabel";
 import { StepIconProps } from "@gemeente-denhaag/stepicon";
 import BaseProps from "@gemeente-denhaag/baseprops";
+// import BaseProps from "@gemeente-denhaag/baseprops";
```
Comment (joostvanviegen) [inline src/components/StepLabel/src/index.tsx:5]: Remove this commented line?

**Review by joostvanviegen — APPROVED** [review]
I approve of these fixes, i do think however that we should remove the lines which have been commented out (instead of being removed) to keep things tidy.

**Review by mjcarsjens — APPROVED** [review]
I approve these changes, once Joost's comments have been resolved this can be merged imo

File: .storybook/main.js (hunk 4 above)
Comment (GewoonMaarten) [reply .storybook/main.js:5]: 3 and 5 are used. 4 is not, I will remove it

File: lerna.json (hunk 5 above)
Comment (GewoonMaarten) [reply lerna.json:7]: no, I think this was a mistake

**Pull request merged**